fw/Core/Hitcon/Logic/IrController.cc

```diff
@@ -175,8 +178,14 @@ void IrController::MaintainQueued() {
     } else if (current_status == kRetransmitStatusWaitTxSlot) {
       // Waiting for IrController's tx slot to open up. (Hash is ready)
       if (current_tx_slot == -1) {
-        bool ret = irLogic.SendPacket(&(queued_packets_[i].data[0]),
-                                      queued_packets_[i].size);
+        bool ret;
+        if (g_xboard_logic.GetConnectState() == UsartConnectState::Connect) {
```

Comment:
We need to tell if we're connected to another badge or connected to the base station.